src/compiler/checker.ts

```diff
@@ -3302,7 +3302,7 @@ namespace ts {
                 function writeTypeReference(type: TypeReference, flags: TypeFormatFlags) {
                     const typeArguments = type.typeArguments || emptyArray;
                     if (type.target === globalArrayType && !(flags & TypeFormatFlags.WriteArrayAsGenericType)) {
-                        writeType(typeArguments[0], TypeFormatFlags.InElementType);
+                        writeType(typeArguments[0], TypeFormatFlags.InElementType | TypeFormatFlags.InArrayType);

```

Comment:
@sandersn `InElementType` is applied to both array and union and intersection contexts. `InArrayType` is just an array context. I could change `InElementType` to be `InUnionOrIntersection` to remove the overlap, but then I need to go through and figure out what places need `InUnionOrIntersection` vs `InArrayType` or both. TBQH it's just easier to leave it as-is, given how the contexts get used.